src/query/functions-v2/src/scalars/boolean.rs

```diff
@@ -145,9 +145,11 @@ pub fn register(registry: &mut FunctionRegistry) {
             }
             None
         },
-        // value = lhs | rhs,  valid = (lhs_v & rhs_v) | (lhs | rhs))
+        // value = lhs | rhs,  valid = (lhs_v & rhs_v) | (lhs_v & lhs) | (rhs_v & rhs)
         vectorize_2_arg::<NullableType<BooleanType>, NullableType<BooleanType>, NullableType<BooleanType>>(|lhs, rhs, _| {
-            let valid = (lhs.is_some() & rhs.is_some()) | (lhs.unwrap_or_default() | rhs.unwrap_or_default());
+            let lhs_v = lhs.is_some();
```

Comment:
```
match (lhs, rhs) {
  
   Some(true), _ => Some(true),
   _, Some(true)  => Some(true),

  Some(false), Some(false) => Some(false),

  _ => None

}
```

Reply:
<img width="275" alt="image" src="https://user-images.githubusercontent.com/3325189/193256235-2ff58e16-d46d-4ff9-8c75-512ed60cca65.png">

Reply:
I think my codes are the same with this?

Reply:
I'm guessing @sundy-li is talking about readability.

Reply:
I think my codes are the same with this?